docs/docs/mail.md

```diff
@@ -1,345 +1,186 @@
-# Mail
+# Conclave Mail
 
-A mail is an authenticated byte array with an encrypted body and a cleartext envelope. The mail can be up to two
-gigabytes in size, however, it must at this time fit into memory when loaded all at once. In practice mails will
-usually be much smaller than that. 
+Conclave uses *Conclave Mail*, a communication mechanism that allows you to send encrypted messages to enclaves that 
+can be decrypted only by enclaves that you trust. Conclave Mail ensures secure communication between a client and an 
+enclave.
 
-## Mail features
+## How Conclave Mail works
```

Comment:
I'm not sure I agree with starting by describing how Mail works. There is a lot of technical detail here, which is not necessarily relevant for a Conclave user who wants to first understand what Mail is.

Reply:
I have split this section into two to reduce the cognitive load on the user. The first section covers the components of a Mail item, and the second covers the internal working. 

As this doc is more of a reference material than a tutorial, I believe it's better to have these sections before describing the features of Conclave. The internal workings section adds more context to the features section and other sections below.